src/field/base/OdeArray.hh

```diff
@@ -0,0 +1,197 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file OdePointers.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Array.hh"
+#include "base/Range.hh"
+#include "base/Types.hh"
+
+#include <cmath>
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * A state array of the equation of motion: pos = v_[0-2], mom = v_[3-5]
+ *
+ * XXX TODO: extend this to include time and spin or to template with a
+ * variable ode_dim and array type
+ */
+
+class OdeArray
+{
+    static constexpr size_type ode_dim = 6;
+    using ode_type                     = Array<real_type, ode_dim>;
+
+  private:
+    ode_type v_;
+
+  public:
+    // Constructors
+    CELER_FUNCTION
+    OdeArray()
+    {
+        for (auto i : range(ode_dim))
+            v_[i] = 0;
+    }
+
+    CELER_FUNCTION
+    OdeArray(const real_type c)
```

Comment:
This should definitely be a explicit constructor -- otherwise a function that expects an OdeArray could get unintentionally passed an integer/double.

I'm actually not sure why this is needed at all -- in what circumstances do the position and momentum have the same numerical value?

Reply:
I was thinking for the case that the whole array is initialized a large value.  Again it is not essential and the constructor is dropped.